pyiron_atomistics/atomistics/structure/sparse_list.py

```diff
@@ -183,13 +183,16 @@ def __getitem__(self, item):
             if len(item) == 0:
                 ind_list = []
             else:
-                if isinstance(item[0], (int, np.integer)):
-                    ind_list = item
-                elif isinstance(item[0], (bool, np.bool_)):
+                # for some reason isinstance(True, int) == True, so check for bools first explicitly
+                if isinstance(item[0], (bool, np.bool_)):
+                    if len(item) != len(self):
+                        raise IndexError("Length of boolean index does not match indexed list!")
                     ind_list = []
                     for i, bo in enumerate(item):
                         if bo:
                             ind_list.append(i)
+                elif isinstance(item[0], (int, np.integer)):
```

Comment:
Should it also be `np.int_`? (I'm not sure though)

Reply:
`np.int_` is a shorthand for the native int for the current platform, whereas `np.integer` really catches all numpy integer types.  I should be using `numbers.Integral` though, I guess.